com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/transaction/ZigBeeTransactionManager.java

```diff
@@ -696,18 +696,21 @@ private void sendNextTransaction() {
                 // if no transaction was sent, the loop will terminate
             } while (0 < transactionsSent);
 
-            // Loop through all outstanding queues and find the next release time
-            long timeout = Long.MAX_VALUE;
-            for (ZigBeeTransactionQueue queue : sleepyOrDelayedQueues) {
-                long nextTime = queue.getNextReleaseTime();
-                if (nextTime < timeout) {
-                    timeout = nextTime;
-                }
-            }
-            sleepyOrDelayedQueues.clear();
-
-            if (timeout > 0) {
-                startRequeueTimer(timeout);
+            // only start a request timer if there is actual outstanding work
+            if (!sleepyOrDelayedQueues.isEmpty()) {
+	            // Loop through all sleepy / delayed queues and find the next release time
+	            long timeout = Long.MAX_VALUE;
+	            for (ZigBeeTransactionQueue queue : sleepyOrDelayedQueues) {
+	                long nextTime = queue.getNextReleaseTime();
```

Comment:
Do we have to care of eventually empty queues?
If not, the test have to changed and we could ignore these queues already instead of adding them to the `sleepyOrDelayedQueues` list.